org.eclipse.xtext.common.types.ui/src/org/eclipse/xtext/common/types/ui/notification/JavaBuilderState.java

```diff
@@ -143,7 +144,13 @@ protected TypeNames _getQualifiedTypeNames(IPackageFragment packageFragment) {
 		String packageName = packageFragment.getElementName();
 		IPath packagePath = resource.getProjectRelativePath();
 		int srcPathSegmentCount = getPackageFragmentRoot(packageFragment).getResource().getProjectRelativePath().segmentCount();
-		for (Object typeLocator : references.keyTable) {
+		Iterable<?> keys;
+		if (references instanceof Map) {
+			keys = ((Map<?,?>) references).keySet();
+		} else {
+			keys = Arrays.asList(((SimpleLookupTable)references).keyTable);
+		}
+		for (Object typeLocator : keys) {
```

Comment:
I was trying to understand what this code is supposed to do with the type locators, but failed until now to grasp it. So I appreciate the local change.